contracts/DomainRegistry.sol

```diff
@@ -0,0 +1,121 @@
+        require(hasValidParentDomain(cleanDomainName), "Parent domain must exist");
+        require(domains[cleanDomainName].controller == address(0), "Domain already reserved");
+
+        domains[cleanDomainName] = Domain({
+            controller: msg.sender,
+            deposit: msg.value 
+        });
+        totalReservedDomains++;
+
+        emit DomainReserved(cleanDomainName, msg.sender, msg.value);
+    }
+
+    function reserveDomainOnlyAssymbly(string memory domainName) public payable  {
+        string memory cleanDomainName = DomainValidator.stripProtocolOnlyAssembly(domainName);
+        require(DomainValidator.isValidDomain(cleanDomainName), "Invalid domain name format");
+        require(hasValidParentDomain(cleanDomainName), "Parent domain must exist");
+        require(domains[cleanDomainName].controller == address(0), "Domain already reserved");
+
+        domains[cleanDomainName] = Domain({
+            controller: msg.sender,
+            deposit: msg.value 
+        });
+        totalReservedDomains++;
+
+        emit DomainReserved(cleanDomainName, msg.sender, msg.value);
+    }
+
+    function hasValidParentDomain(string memory domainName) private view returns (bool) {
+        string memory parentDomain = DomainValidator.getParentDomain(domainName);
+        if(bytes(parentDomain).length == 0) {
+            return true;
+        }
+        return domains[parentDomain].controller != address(0);
+    }
+
+
+    function isDomainRegistered(string memory domainName) external view returns (bool) {
+        string memory cleanDomainName = DomainValidator.stripProtocol(domainName);
+        return domains[cleanDomainName].controller != address(0);
+    }
+
+    function changeDeposit(string memory domainName, uint256 newDeposit) public payable onlyController(domainName) sufficientDeposit(newDeposit) {
+        string memory cleanDomainName = DomainValidator.stripProtocol(domainName);
+        domains[cleanDomainName].deposit = msg.value;
+        emit DepositChanged(cleanDomainName, msg.value);
+    }
+
+    function transferDomainControl(string memory domainName, address newController) public onlyController(domainName) {
+        string memory cleanDomainName = DomainValidator.stripProtocol(domainName);
+        require(newController != address(0), "Invalid controller address");
+
+        domains[cleanDomainName].controller = newController;
+        emit DomainControlTransferred(cleanDomainName, newController);
+    }
+
+    function releaseDomain(string memory domainName) public onlyController(domainName) {
+        string memory cleanDomainName = DomainValidator.stripProtocol(domainName);
+        address payable controller = payable(msg.sender);
+        uint256 refundAmount = domains[cleanDomainName].deposit;
+
+        delete domains[cleanDomainName];
+        totalReservedDomains--;
+        controller.transfer(refundAmount);
+
+        emit DomainReleased(domainName, msg.sender, refundAmount);
+    }
```

Comment:
You don't check if the domain has child domains before deleting it. Try to reorganise the the data and the code.